task/sensor/src/main.rs

```diff
@@ -251,6 +317,10 @@ fn main() -> ! {
         last_reading: SensorArray(last_reading),
         data_value: SensorArray(data_value),
         data_time: SensorArray(data_time),
+        min_value: SensorArray(min_value),
+        min_time: SensorArray(min_time),
+        max_value: SensorArray(max_value),
+        max_time: SensorArray(max_time),
```

Comment:
This is getting silly enough that a macro may be worthwhile:
```rust
    macro_rules! sensor_array {
        ($($name:ident: $t:ty = $n:expr;)*) => {
            paste::paste! {
                let ($($name),*) = mutable_statics::mutable_statics! {
                    $(
                    static mut [<$name:upper>]: [$t; NUM_SENSORS] = [|| $n; _];
                    )*
                };
                let ($($name),*) = ($(SensorArray($name)),*);
            }
        };
    }
    sensor_array!(
        last_reading: Option<LastReading> = None;
        data_value: f32 = f32::NAN;
        data_time: u64 = 0u64;
        min_value: f32 = f32::MAX;
        min_time: u64 = 0u64;
        max_value: f32 = f32::MIN;
        max_time: u64 = 0u64;
        err_value: NoData = NoData::DeviceUnavailable;
        err_time: u64 = 0;
        nerrors: u32 = 0;
    );

    let mut server = ServerImpl {
        last_reading,
        data_value,
        data_time,
        min_value,
        min_time,
        max_value,
        max_time,
        err_value,
        err_time,
        nerrors,
        deadline,
    };
```
(this requires adding `paste.workspace = true` to the `sensors/Cargo.toml`)

I haven't tested it, but it compiles fine and has the exact same size in flash / RAM.

Reply:
Or even
```rust
    macro_rules! declare_server {
        ($($name:ident: $t:ty = $n:expr;)*) => {{
            paste::paste! {
                let ($($name),*) = mutable_statics::mutable_statics! {
                    $(
                    static mut [<$name:upper>]: [$t; NUM_SENSORS] = [|| $n; _];
                    )*
                };
                let ($($name),*) = ($(SensorArray($name)),*);
                ServerImpl {
                    deadline,
                    $($name),*
                }
            }}
        };
    }
    let mut server = declare_server!(
        last_reading: Option<LastReading> = None;
        data_value: f32 = f32::NAN;
        data_time: u64 = 0u64;
        min_value: f32 = f32::MAX;
        min_time: u64 = 0u64;
        max_value: f32 = f32::MIN;
        max_time: u64 = 0u64;
        err_value: NoData = NoData::DeviceUnavailable;
        err_time: u64 = 0;
        nerrors: u32 = 0;
    );
```
(which removes one more layer of duplication, at the cost of some clarity)

Reply:
That looks great! I have verified that it works; pulling it in.